tools/assetpipelineeditor/dracocompressor.cpp

```diff
@@ -0,0 +1,175 @@
+
+    QVector<T> indicesVector;
+    indicesVector.resize(attribute.count());
+
+    if (byteStride == 0)
+        for (int i = 0, nbIndices = indicesVector.size(); i < nbIndices; ++i)
+            indicesVector[i] = static_cast<T>(data.data()[i * sizeof(T)]);
+    else
+        for (int i = 0, nbIndices = indicesVector.size(); i < nbIndices; ++i)
+            indicesVector[i] = static_cast<T>(data.data()[i * byteStride]);
+
+    return indicesVector;
+}
+
+template<typename T>
+void addFacesToMesh(const QVector<T> &indices, draco::Mesh &dracoMesh)
+{
+    for (int i, nbIndices = indices.size() / 3; i < nbIndices; ++i)
+        dracoMesh.AddFace({ static_cast<draco::IndexType<unsigned int, draco::PointIndex_tag_type_>>(indices[3 * i + 0]),
+                            static_cast<draco::IndexType<unsigned int, draco::PointIndex_tag_type_>>(indices[3 * i + 1]),
+                            static_cast<draco::IndexType<unsigned int, draco::PointIndex_tag_type_>>(indices[3 * i + 2]) });
+}
+} // namespace
+
+std::unique_ptr<draco::EncoderBuffer> compressMesh(const Qt3DRender::QGeometry &geometry)
+{
+    std::unique_ptr<draco::EncoderBuffer> compressBuffer = std::unique_ptr<draco::EncoderBuffer>(new draco::EncoderBuffer);
+    draco::Mesh dracoMesh;
+
+    std::unordered_map<Qt3DRender::QBuffer *, draco::DataBuffer *> attributeBufferToDataBuffer;
+    // Convert Qt3D geometry to draco mesh
+    const auto geometryAttributes = geometry.attributes();
+    Qt3DRender::QAttribute *indicesAttribute = nullptr;
+    for (const auto attribute : geometryAttributes) {
+
+        const auto dracoAttributeType = attributeTypeFromName(attribute->name());
+
+        if (attribute->attributeType() == Qt3DRender::QAttribute::IndexAttribute)
+            indicesAttribute = attribute;
+
+        const auto dracoDataType = dataTypeFromVertexBaseType(attribute->vertexBaseType());
+        if (dracoAttributeType == draco::GeometryAttribute::Type::INVALID ||
+            dracoDataType == draco::DataType::DT_INVALID)
+            continue;
+
+        const auto attributeBuffer = attribute->buffer();
+        if (attributeBufferToDataBuffer.find(attributeBuffer) == std::end(attributeBufferToDataBuffer))
+            attributeBufferToDataBuffer[attributeBuffer] = createDracoBufferFromQBuffer(*attributeBuffer);
+
+        const auto dracoBuffer = attributeBufferToDataBuffer[attributeBuffer];
+        auto data = reinterpret_cast<float *>(dracoBuffer->data());
+        qDebug() << "---Init---";
+        for (int i = 0; i < dracoBuffer->data_size() / sizeof(float); ++i)
+            qDebug()
+                    << data[i];
+        qDebug() << "---End----";
+
+        draco::GeometryAttribute meshAttribute;
+        meshAttribute.Init(dracoAttributeType,
+                           dracoBuffer,
+                           attribute->vertexSize(),
+                           dracoDataType,
+                           false,
+                           attribute->byteStride(),
```

Comment:
stride and offset should both be 0 since your copying data into separate new buffers

Reply:
I cant see why. Im creating a copy of the QBuffer into the dracoBuffer, so I think stride and offset should be the same, no?